### DISCUSSION TITLE
Should environment variables be written to `workers` or should the `web` also require them?

### DISCUSSION THREAD

**Discussion by aliuq** [karakeep-app/karakeep#187] [Q&A] [answered]
Hi, I found that openAI requests are being executed repeatedly

**Comment by kamtschatka** [comment]
Depends on the Environment Variable. Which one are you talking about?
You can check them here: https://docs.hoarder.app/configuration/

As a general rule, everything from inference configs downwards should be in the worker.
The upper section is mixed.

What exactly is the issue you are seeing, I am not sure why you are adding "I found that openAI requests are being executed repeatedly" to the question about which environment variable to set where?

**Reply by aliuq** [reply]
Thanks for the reply, I would like to know whether environment variables such as `OPENAI_` and `CRAWLER_` need to be added to the `web` service

![image](https://github.com/hoarder-app/hoarder/assets/11707934/c181ae4d-3245-475a-8579-68b5e6395390)

**Reply by kamtschatka** [reply, marked as answer]
no `OPENAI_`  and `CRAWLER_`  aren't used by the web container

**Reply by aliuq** [reply]
Thank you very much ☺️

This is a great project
